prov/efa/src/rxr/rxr_init.c

```diff
@@ -477,6 +473,11 @@ static int rxr_info_to_rxr(uint32_t version, const struct fi_info *core_info,
 		}
 	}
 
+	/* Use a table for AV if the app has no strong requirement */
+	if (!hints || !hints->domain_attr ||
+	    hints->domain_attr->av_type == FI_AV_UNSPEC)
+		info->domain_attr->av_type = FI_AV_TABLE;
```

Comment:
should here be a else branch to set ```info->domain_attr->av_type = hints->domain_attr->av_type```?

Reply:
av_type is set in fi_alter_domain_attr. I changed fi_getinfo to set FI_AV_MAP and that works.

Reply:
I see. So if hints is provided, there will be a call to ```ofi_alter_info```, which call ```fi_alter_domain_attr``` which set av_type. 

This change apply to the case no hints has been applied.

Can we make it the else branch of the  ```if (hints)``` call?  like

```
if (hints) {
   
} else {
    info->domain_attr->av_type = FI_AV_TABLE;
}
```

Reply:
This check works regardless of whether hints is specified so it's equivalent to what you posted. The only time it's not triggered is if FI_AV_TABLE or FI_AV_MAP is set, which is what we want.

Reply:
Thanks!